### PR TITLE
Added avgas to list of valid fuel choices for gasoline engines

### PR DESCRIPTION
**Pull request by Ilurker**
#### Summary
Balance "Make avgas burnable by cars"

#### Purpose of change
Avgas is trade name for a variety of leaded, high octane gasoline which is usable in standard gasoline engines, but the game does not treat it as a valid fuel type. While avgas certainly isn't good for a typical engine, it's not immediately harmful and wouldn't produce issues across the typical lifespan of a CDDA player, much less their vehicles - much less than burning rendered spider fat would cause a humvee, for instance. The primary variety of avgas in usage today appears to be 110LL, which is much lower lead than typical. There are many varieties of unleaded avgas on and coming to market (UL102 and G100UL), and in the "near future" context with such a relatively high prevalence of solar polar, it is not outlandish to assume that most avgas is of these "green" varieties.

#### Describe the solution

Added avgas to the fuel options for gasoline engines

#### Describe alternatives you've considered

Finding lore reasons for players to adhere to the environmental prohibition on burning leaded fuel in standard engines.
Adding catalytic converters which must be removed to utilize avgas.
Having avgas and other makeshift fuel slowly damage engines.

#### Testing

Spawned a gasoline-powered vehicle and replaced its fuel with avgas. Verified that the vehicle was able to drive with avgas as its only fuel source and that the fuel selection dialogue existed and recognized avgas as an option.

#### Additional context

https://en.wikipedia.org/wiki/Avgas
https://www.ainonline.com/aviation-news/business-aviation/2016-03-30/faa-moves-next-phase-unleaded-avgas-testing
https://web.archive.org/web/20170415200334/https://www.lycoming.com/node/18093
https://bobistheoilguy.com/forums/threads/use-avgas-in-car.19939/
https://classicmotorsports.com/articles/fuel-facts-why-not-avgas-your-car/

### CODE REVIEW

**Comment by Drew4484** [comment]
I do think that the catalytic converter issue is significant, while we don't currently have that as a vehicle part, they're required on American vehicles going back nearly 40 years, meaning they're common enough that most encountered vehicles would have one as part of their exhaust system, meaning they would be negatively impacted by burning avgas. Best case, this is a reduction in fuel efficiency, worst case the vehicle can't properly exhaust and the engine is damaged.

**Comment by Ilurker** [comment]
I recognize it's a long term concern, but I don't think the typical player will be burning enough tanks of avgas to significantly foul their catalytic converters - there's a lot of anecdotal accounts of people using it without wrecking their car.

I'm not completely sure about the timeline we're dealing with for the game right now (I understand there's been some shifting around), but with the first unleaded avgases hitting market in 2016 and becoming approved fuel types by 2017, it could be moot in this universe.

**Comment by anothersimulacrum** [comment]
> I'm not completely sure about the timeline we're dealing with for the game right now (I understand there's been some shifting around), but with the first unleaded avgases hitting market in 2016 and becoming approved fuel types by 2017, it could be moot in this universe.

For questions like that, look at what there is right now.

**Comment by Ilurker** [comment]
The canonical date is 2023, but given that it's a universe with extremely high electric vehicle and small-scale solar (and nuclear!) penetration, I'd be confident in assuming rollouts of other environmental initiatives are also several years ahead. Unleaded avgas is similar to electric in that it exists, it's just low-penetration at present.

eg:

https://www.avweb.com/aviation-news/faa-approves-600-engines-for-gami-unleaded-fuel/

**Comment by anothersimulacrum** [comment]
When in doubt, use current real-world data. 

> given that it's a universe with extremely high electric vehicle and small-scale solar (and nuclear!) penetration

I am not sure this is entirely true (there is no nuclear), and what of it is true does not mean it is correct.

**Comment by Ilurker** [comment]
Atomic-powered household items are cornerstones of a typical survivor loadout:

https://nornagon.github.io/cdda-guide/#/item/wearable_atomic_light
https://nornagon.github.io/cdda-guide/#/item/light_minus_atomic_battery_cell
https://nornagon.github.io/cdda-guide/#/item/atomic_lamp_off
https://nornagon.github.io/cdda-guide/#/item/atomic_coffeepot
https://nornagon.github.io/cdda-guide/#/vehicle_part/atomic_light

**Comment by anothersimulacrum** [comment]
My bad, I forgot about those. But those do not justify a divergence from real data - it's roughly the same argument as the second one [here](https://github.com/CleverRaven/Cataclysm-DDA/wiki/Frequently-Made-Suggestions#failed-rationalizations).

**Comment by Ilurker** [comment]
Some additional discussions. It's difficult to find a good breakdown of avgas usage because it's quite illegal, but it's very clear that:

a) this fuel gets used in automobiles in a post-catalytic converter world
b) their vehicles operate long enough for the people who use it to believe it's a good idea
c) people don't generally believe things are a good idea when they destroy their treasured belongings
d) when treasured belongings do get destroyed, people talk about their experiences. There are no first hand accounts of people using avgas and ruining their engines on hand, which you'd expect to see in discussions between people who use it

https://www.snowest.com/forum/threads/av-gas-vs-race-fuel.398529/
https://www.thumpertalk.com/forums/topic/1186807-anyone-running-avgas-vs-race-fuel/#comments
https://www.thumpertalk.com/forums/topic/472662-what-to-tell-the-people-at-the-airport-when-you-get-av-gas/
https://www.thumpertalk.com/forums/topic/1002592-anyone-run-avgas-100ll/
https://jalopnik.com/why-you-really-shouldnt-run-jet-fuel-in-your-car-5938552
https://www.clubgti.com/forums/index.php?threads/how-do-i-run-avgas-in-my-car.211952/
https://www.quora.com/What-happens-if-an-airplane-fuel-is-used-on-normal-automobiles
https://www.fordmuscleforums.com/threads/anybody-run-avgas-ll100.508143/

Basically, the discussion is divided between people who are sure it would destroy your engine immediately, and people who at least claim to be using it just fine. I'd heavily tend to agree with copious personal testimony, here - there are lots of warnings about what avgas **might do** to your vehicle's lifespan, but very few people talking about what avgas **did do** to their vehicle. You'd expect a much higher ratio of cautionary tales if its effects were pronounced and immediate.

The reality is that avgas makes your car go - folk are talking about putting it in their cars and making it go, and there is no debate whether or not it's gasoline. The blocker at hand is whether one form of realism (that this fuel will, absolutely, make your car run for long enough that guys who race cars think it's a good idea to use) is outweighed by another (that CDDA doesn't model vehicle operational wear at tear in any situation, ever).

My concern is that this logic could be applied to block pretty much anything. You may as well say you can't, say, add a new vehicle appliance because you aren't modeling what moving its weight to one side of the vehicle will do to your suspension over the next six months. Or that you can't add new illumination because you don't model overloading your electronics.

tl;dr Allowing vehicles to burn avgas will undoubtedly increase one parameter of realism - that gasoline should be usable in gasoline engines - while any given harm to realism is predicated on the implementation of an engine wear system which doesn't exist and therefore already confers no realism.

**Comment by kevingranade** [comment]
I am seeing a lot of specious arguments here and nothing compelling so I'm gonna say no.
